### PR TITLE
Add E2E reproduction for #39150

### LINKED ISSUE #39150: Custom column with a duplicate name fails in nested models
### Describe the bug

When you create a custom column, giving it the same name as an existing column fails.

### To Reproduce

3 models, each model has the same custom column name (first model is a simple one, no custom column)

Screenshot 2024-02-23 at 8 27 38 PM
Screenshot 2024-02-23 at 8 28 05 PM
Screenshot 2024-02-23 at 8 28 42 PM

### Expected behavior

_No response_

### Logs

_No response_

### Information about your Metabase installation

```JSON
48.6

https://vamsi-peri.hosted.staging.metabase.com/model/102-accounts-from-fb-google-v3-model
```

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Reproduces #39150 as a confirmation that the issue was fixed.
Closes #39150.

### Stress-test
20x ✅ 
https://github.com/metabase/metabase/actions/runs/9372789466/job/25805089064

Another stress test with the two repro variants after df54dd15501cff1b2409ebd6bec108c329854c96  (20x ✅ )
https://github.com/metabase/metabase/actions/runs/9389943859

### CODE REVIEW

**Review by romeovs — APPROVED** [review]
LGTM, just one small nit.

File: e2e/test/scenarios/models/reproductions.cy.spec.ts [hunk 1]
```diff
@@ -289,3 +292,67 @@ describe("issue 23103", () => {
...
+describe("issue 39150", () => {
+  const ccName = "CC Rating";
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it(
+    "shows allows custom columns with the same name in nested models (metabase#39150)",
```
Comment (kamilmielnik) [inline e2e/test/scenarios/models/reproductions.cy.spec.ts:305]:
```suggestion
    "allows custom columns with the same name in nested models (metabase#39150)",
```

File: e2e/test/scenarios/models/reproductions.cy.spec.ts [hunk 2]
```diff
@@ -289,3 +292,67 @@ describe("issue 23103", () => {
...
+    { viewportWidth: 1600 },
```
Comment (kamilmielnik) [inline e2e/test/scenarios/models/reproductions.cy.spec.ts:306]: Another way to deal with too many columns on the screen is to specify `fields` in question query and use just the columns you need - there's a small performance benefit to it.

File: e2e/test/scenarios/models/reproductions.cy.spec.ts [hunk 3]
```diff
@@ -289,3 +292,67 @@ describe("issue 23103", () => {
...
+    () => {
+      createQuestion({
```
Comment (kamilmielnik) [inline e2e/test/scenarios/models/reproductions.cy.spec.ts:308]: I think there are 2 slightly different ways to reproduce the issue and they both deserve a test:
- custom column being named like an aggregation column in source question (original repro in [#39150](https://github.com/metabase/metabase/issues/39150))
- custom column being named like a custom column in source question ([comment](https://github.com/metabase/metabase/issues/39150#issuecomment-2029796451))

Comment (nemanjaglumac) [reply e2e/test/scenarios/models/reproductions.cy.spec.ts:308]: The issue was ambiguous to me with poor repro steps, and I didn't interpret the first case as the "custom column being named like an aggregation column in source question" but I can add that for sure.

File: e2e/test/scenarios/models/reproductions.cy.spec.ts [hunk 4]
```diff
@@ -289,3 +292,67 @@ describe("issue 23103", () => {
...
+        name: ccName,
+        blur: true,
+      });
+
+      cy.button("Done").click();
+
+      visualize();
+
+      cy.findAllByTestId("header-cell")
+        .filter(`:contains('${ccName}')`)
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply e2e/test/scenarios/models/reproductions.cy.spec.ts:354]: I'll double check. 
Thanks

File: e2e/test/scenarios/models/reproductions.cy.spec.ts (hunk 2 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/models/reproductions.cy.spec.ts:306]: I am aware.
But this is less lines of code to read :D

And I'm already limiting results to two rows so we shouldn't have problems with the performance.

File: e2e/test/scenarios/models/reproductions.cy.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply e2e/test/scenarios/models/reproductions.cy.spec.ts:354]: The answer is no.
image

**Review by ranquild — APPROVED** [review]

**Comment by uladzimirdev** [comment]
I just realized we had a repro already https://github.com/metabase/metabase/pull/40851

**Pull request merged**